plugins/curation/server/__init__.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2016 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+from girder.api import access
+from girder.api.describe import Description, describeRoute
+from girder.api.rest import Resource, loadmodel
+from girder.constants import AccessType
+from girder.utility import mail_utils
+import datetime
+import posixpath
+
+
+CURATION = 'curation'
+
+ENABLED = 'enabled'
+STATUS = 'status'
+TIMELINE = 'timeline'
+ENABLE_USER_ID = 'enableUserId'
+REQUEST_USER_ID = 'requestUserId'
+REVIEW_USER_ID = 'reviewUserId'
+
+CONSTRUCTION = 'construction'
+REQUESTED = 'requested'
+APPROVED = 'approved'
+
+DEFAULTS = {
+    ENABLED: False,
+    STATUS: CONSTRUCTION,
+}
+
+
+class CuratedFolder(Resource):
+
+    @access.public
+    @loadmodel(model='folder', level=AccessType.READ)
+    @describeRoute(
+        Description('Get curation details for the folder.')
+        .param('id', 'The folder ID', paramType='path')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Read permission denied on the folder.', 403)
+    )
+    def getCuration(self, folder, params):
+        result = dict(DEFAULTS)
+        result[TIMELINE] = []
+        result.update(folder.get(CURATION, {}))
+        return result
+
+    @access.public
+    @loadmodel(model='folder', level=AccessType.READ)
+    @describeRoute(
+        Description('Set curation details for the folder.')
+        .param('id', 'The folder ID', paramType='path')
+        .param('enabled', 'Enable or disable folder curation.',
+               required=False, dataType='boolean')
+        .param('status', 'Set the folder curation status.',
+               required=False, dataType='string')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Write permission denied on the folder.', 403)
+    )
+    def setCuration(self, folder, params):
+        user = self.getCurrentUser()
+        if CURATION not in folder:
```

Comment:
not that you need to change it, but I recently found `setdefault` and thought it was useful

it would allow you to replace

    if CURATION not in folder:
        folder[CURATION] = dict(DEFAULTS)
    curation = folder[CURATION]

with

    curation = folder.setdefault(CURATION, dict(DEFAULTS))